core/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -321,8 +321,11 @@ public T get(Settings settings) {
     public String getRaw(Settings settings) {
         // They're using the setting, so we need to tell them to stop
         if (this.isDeprecated() && this.exists(settings)) {
+            if (deprecationLogger.get() == null) {
```

Comment:
ok. Considering `Setting` as a simple value class, it should probably have no logger at all (but that's a whole different story).